src/extension/commandPaletteHandler.ts

```diff
@@ -111,13 +119,16 @@ export class CommandPaletteHandler {
     public static publishToExpHost(): Q.Promise<void> {
         return this.selectProject()
             .then((project: IReactNativeProject) => {
-                return this.executeCommandInContext("publishToExpHost", project.workspaceFolder, () => {
-                    return this.executePublishToExpHost(project).then((didPublish) => {
-                        if (!didPublish) {
-                            CommandPaletteHandler.logger.warning(localize("ExponentPublishingWasUnsuccessfulMakeSureYoureLoggedInToExpo", "Publishing was unsuccessful. Please make sure you are logged in Expo and your project is a valid Expo project"));
-                        }
+                return CommandPaletteHandler.checkReactNativePackageExistence(project.workspaceFolder.uri.path)
```

Comment:
Please make sure that this check is required here because Expo may have a non-standard setup.

Reply:
Also I'm not sure if we need that here since this is a publish function.

Reply:
I rolled back this change, because there is already version verification in `runExponent` function.